pgrx/src/fcinfo.rs

```diff
@@ -332,30 +335,30 @@ unsafe fn direct_function_call_as_datum_internal(
     args: &[Option<pg_sys::Datum>],
 ) -> Option<pg_sys::Datum> {
     let nargs: i16 = args.len().try_into().expect("too many args passed to function");
-    let fcinfo_ptr = pg_sys::palloc(
+    let fcinfo = pg_sys::palloc0(
         std::mem::size_of::<pg_sys::FunctionCallInfoBaseData>()
             + std::mem::size_of::<pg_sys::NullableDatum>() * args.len(),
     )
     .cast::<pg_sys::FunctionCallInfoBaseData>();
 
-    let fcinfo = fcinfo_ptr.as_mut().unwrap_unchecked();
-    fcinfo.flinfo = std::ptr::null_mut();
-    fcinfo.context = std::ptr::null_mut();
-    fcinfo.resultinfo = std::ptr::null_mut();
-    fcinfo.fncollation = pg_sys::InvalidOid;
-    fcinfo.isnull = false;
-    fcinfo.nargs = nargs;
+    (*fcinfo).flinfo = std::ptr::null_mut();
+    (*fcinfo).context = std::ptr::null_mut();
+    (*fcinfo).resultinfo = std::ptr::null_mut();
+    (*fcinfo).fncollation = pg_sys::InvalidOid;
+    (*fcinfo).isnull = false;
+    (*fcinfo).nargs = nargs;
 
-    let arg_slice = fcinfo.args.as_mut_slice(args.len());
```

Comment:
This? Using a palloc, which creates uninit memory, and taking `&mut` to an initialized type? Worse, one with illegal bitpatterns (bool)? This was simply UB. We could do `&mut [MaybeUninit<NullableDatum>]`, but not this.